### PR TITLE
3 feature 3 create review

### PR DESCRIPTION
**Pull request by jkahn506**
Added logic to Review Usecases
Added Serialization for AverageRatings and ReviewDatabase
Created a controller class for Review

**Note**
I had to change the entities ReviewDatabase and AverageRatings a bit to make serialization work. You'll have to create specific instances of them to use their get methods. To properly load an instance of ReviewDatabase, use the static method ReviewInteractor.loadReviewDatabase(). To properly load an instance of AverageRatings, use the static method UpdateAverageRating.loadRatingDatabase(). Sorry for the difficulties!

### CODE REVIEW

File: src/main/java/review/ReviewInteractor.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    private ReviewDatabase reviewDatabase;
+    private static final ReviewDatabaseReadWriter databaseReadWriter = new ReviewDatabaseReadWriter();
+
+
+    /**
+     * Construct a ReviewInteractor.
+     */
+    public ReviewInteractor() {
+        try {
+            RecipeRepoGateway rrg = new RecipeRepoImpl();
```
Comment (Hclover2003) [inline src/main/java/review/ReviewInteractor.java:21]: RecipeRepoImpl is now RecipeReadWriter --> need to call RecipeReadWriter.getAllRecipes()

File: src/main/java/review/UpdateAverageRating.java [hunk 2]
```diff
@@ -1,4 +1,54 @@
...
+        AverageRatingReadWriter ratingReadWriter = new AverageRatingReadWriter();
+        try {
+            database = ratingReadWriter.readFromFile("ratings.sav");
+        } catch (IOException e) {
+            database = new AverageRatings();
+            System.out.println("Read file failed.....");
+        } catch (ClassNotFoundException e) {
+            throw new RuntimeException(e);
+        }
+        return database;
```
Comment (Hclover2003) [inline src/main/java/review/UpdateAverageRating.java:52]: looks good! :D

File: src/test/java/entities/AverageRatingsTest.java [hunk 3]
```diff
@@ -7,23 +7,25 @@
 
 class AverageRatingsTest {
 
+    static AverageRatings r = new AverageRatings();
```
Comment (Hclover2003) [inline src/test/java/entities/AverageRatingsTest.java:10]: maybe rename this to something that gives some more information? ex. averageRatings

**Review by Hclover2003 — APPROVED** [review]
Looks good!
- changed static class to non-static class, and carried this over by creating instance variables in the respective files
- small formatting changes - maybe rename variables to make sure they provide enough info

**Review by YadingF — APPROVED** [review]
ok

File: src/main/java/review/ReviewInteractor.java [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+import recipe.RecipeRepoGateway;
+import recipe.RecipeRepoImpl;
+
+import java.io.IOException;
+
+public class ReviewInteractor {
+
+    private RecipeList recipeList;
+    private ReviewDatabase reviewDatabase;
+    private static final ReviewDatabaseReadWriter databaseReadWriter = new ReviewDatabaseReadWriter();
```
Comment (jkahn506) [inline src/main/java/review/ReviewInteractor.java:13]: I should probably utilize the dependency injection design pattern here

**Review by Brenden-Yiping-Wang — APPROVED** [review]
The change is great. In this way, we can interact with the reviews.

**Pull request merged**